changes/tasks.py

```diff
@@ -1468,6 +1468,88 @@ def run(self, vumi_message_id: str, event_type: str, **kwargs) -> None:
 process_whatsapp_system_event = ProcessWhatsAppSystemEvent()
 
 
+class ProcessWhatsAppTimeoutSystemEvent(Task):
+    """
+    Notifies a user that a message we send them on Whatsapp has not been delivered
+    within 7 days, she will be sent an outbound
+    message informing them and give the option to switch to SMS.
+    """
+
+    name = "ndoh_hub.changes.tasks.process_whatsapp_timeout_system_event"
+
+    def handle_undelivered(self, user_id, source_id, identity_uuid):
+        identity = is_client.get_identity(identity_uuid)
+        details = identity["details"]
+        if "timeout_timestamp" not in details:
+            identity["details"]["timeout_timestamp"] = datetime.today()
+            # Transform to django language code
+            language = identity["details"]["lang_code"].lower().replace("_", "-")
+            with translation.override(language):
+                text = translation.ugettext(
+                    "We see that your MomConnect WhatsApp messages are not being "
+                    "delivered. If you would like to receive your messages over "
+                    "SMS, reply ‘SMS’."
+                )
+
+                utils.ms_client.create_outbound(
+                    {
+                        "to_identity": identity_uuid,
+                        "content": text,
+                        "channel": "JUNE_TEXT",
+                        "metadata": {},
+                    }
+                )
+        else:
+            d1 = datetime.fromtimestamp(details["timeout_timestamp"])
+            d2 = datetime.today()
+            week1 = d1 - timedelta(days=d1.weekday())
+            week2 = d2 - timedelta(days=d2.weekday())
+            # Returns 0 if both dates fall withing one week, 1 if on two weeks etc.
+            weeks = int(round((week2 - week1).days / 7))
+            months = int(round((week2 - week1).days / 30))
+            if weeks == 1 and months == 0:
```

Comment:
We want this to be a month by default, but also it would be nice for this to be configurable.

This logic can also be simplified quite a bit. When you subtract two datetime objects, you get a `timedelta` object of the difference between the two. https://docs.python.org/3/library/datetime.html#timedelta-objects

We can also be using `datetime.now` instead of `datetime.today` to get a more accurate time difference, but we should also probably also ensure that we're using timezone aware timestamps, so that the time comparison is more accurate.

So we could do something like:
```
if (d2 - d1).days >= settings.WHATSAPP_EXPIRY_SMS_BOUNCE_DAYS:
    ....
```

Reply:
am I able to add 2 configurable dates in `WHATSAPP_EXPIRY_SMS_BOUNCE_DAYS` in settings, adding that it should be a week, but if it is within 7+ days but less than a month it shouldn't send?